examples/step-5/step-5.cc

```diff
@@ -113,6 +113,14 @@ double coefficient (const Point<dim> &p)
   else
     return 1;
 }
+// @note It looks like we use a discontinuous coefficient, however,
+// the finite element method using quadrature really sees only a
+// polynomial approximation (defined by a
+// <code>quadrature_formula</code>) of that discontinuous coefficient
+// on every mesh. (Though that polynomial approximation of course
+// tends to a discontinuous function under mesh refinement.) Refer to
+// CellAccessor::set_material_id to set a discontinuous coefficient
```

Comment:
I would say then that this discussion probably on using `material_id`'s is probably best left to `step-6` where one discusses how to make a better / more representative mesh.
There one has the picture with which one can fully explain the problem at hand.
Furthermore, `step-5` is the first time that one encounters non-constant coefficients, and it would be best not to unnecessarily overcomplicate the tutorial.

What we could do, however, is make a note of the approximation of the discontinuous material properties just where assembly is done. What do you think of something like this?
```
// Next is the typical loop over all cells to compute local contributions 
// and then to transfer them into the global matrix and vector. The only 
// change in this part, compared to step-4, is that we will use the 
// <code>coefficient</code> function defined above to compute the 
// coefficient value at each quadrature point.
//
// Note that for cell's whose quadrature points evaluate to both values of the 
// material coefficient, the coefficient as approximated on the cell is not 
// strictly discontinuous. Rather, a local approximation of the coefficient 
// is formed based on both the chosen <code>finite_element</code> and 
// <code>quadrature_formula</code>. This polynomial approximation 
// of course tends towoards the discontinuous function under mesh refinement.
```

If you're happy with that then we could focus more on the discussion of the interior interface between material zones at the end of `A better mesh` in `step-6`.